driver/src/main/scala/core/nodeset.scala

```diff
@@ -344,4 +354,12 @@ class ChannelFactory(options: MongoConnectionOptions, bossExecutor: Executor = E
     config.setConnectTimeoutMillis(options.connectTimeoutMS)
     channel
   }
+
+  private object TrustAnyManager extends javax.net.ssl.X509TrustManager {
+    import java.security.cert.X509Certificate
+
+    override def checkClientTrusted(cs: Array[X509Certificate], a: String) = {}
```

Comment:
I guess it's for your tests but we should avoid that.
The thing we can do is to add a config key to skip the certificate check (obviously set to false by default).